CONTRIBUTING.md

```diff
@@ -66,110 +62,133 @@ info
+When a new component is being created, work with the design team to document design and usage guidelines in the component folder's README.md file.
 
-### Introducing 3rd Party Components
+#### Consider the Following Before Adding a New Component
 
-There are 3 ways to introduce 3rd party components to the `UI-kit`
+**Is it likely this component be used in more than 1 app?**
+For example, a component for a card that may contain any content would be a good candidate for ui-kit. A card that specifically renders data specific to an object in an napp is not.
 
-* Link  
-  We add a link to that component to the documentation. This is not a favorable way.
-* Wrap  
-  We are creating our own component but in that component we are utilising a 3rd party component, this solution is the favourable solution for the `UI-kit` as with that solution we are adding consistency to the 3rd party components and there is a single point to change the usage of that component if we decide to update the component or replace the component with our own implementation or the other implementation.
-* Fork  
-   We should do that as a last resort if we do not see any other solution.
+**Does it fit in with a "theme" of other components in the UI Kit?**
+For example, if we have a toast and an inline alert box, a page banner would make sense to exist in the ui-kit
 
-<!--
-TODO: add step by step guide for forking.
-- Key decisions needed
-    - we forking repositories into the UI-kit or are we creating a new repository within `dcos-labs`.
--->
+**Is the effort that goes into building and maintaining it as a reusable component worth the payoff?**
+For example, Table and it's related components are complex components to properly maintain, but it's such a heavily used pattern in dcos-ui, that it's worth the effort
 
-Going from link via wrap to fork increases the control over component but also increases the effort need to update the component. A wrap is a sweet spot between consistency and functionality vs control and effort.
+**Will side effects of adding this component help us move the design system forward?**
+For example, adding a toast component forces us to think about visual indication of alert status (error, success, warning, etc.), z-index stacking context, accessibility guidelines, and how we animate a piece of UI in and out of the viewport.
 
-### Moving A Component From `dcos/dcos-ui` To The `ui-kit`
+**Is now the right time to add this to UI Kit?**
+If adding this component right now prevent us from working on more important work, maybe it should be built in dcos-ui, and we’ll follow-up about adding to UI Kit at the end of the release iteration.
+If I can’t get this added to UI Kit in time for it to be used in dcos-ui to resolve your feature/task/bug, build in dcos-ui, and we’ll follow-up about adding to UI Kit at the end of the release iteration.
```

Comment:
i think we don't need to mention dcos-ui in here.

Reply:
You're absolutely right, my mistake. I copied that paragraph from something I wrote in 2018 😅